jobs/test_e2e_errand/templates/bin/run.erb

```diff
@@ -15,40 +16,63 @@ export PATH="/var/vcap/packages/java8/bin:$PATH"
 
 [ ! -e /home/vcap/.sincedb ] || rm /home/vcap/.sincedb
 
-
-echo '' ; echo '==> Purging existing test data...'
-
-curl -s -X DELETE '<%= p('test_e2e_errand.elasticsearch_hostport') %>/_all/sample-logs--repo-commits' ; echo ''
-
+export QUERY=$(cat <<EOF
+{ "query": {
+    "filtered" : {
+        "query" : {
+            "term" : { "@source.host":"sample-logs--repo-commits" }
+        },
+        "filter" : {
+          "bool" : {
+            "must" : {
+                "range" : {
+                    "@timestamp" : {
+                        "from" : "$(date --iso-8601=ns)",
+                        "to" : "now"
+                    }
+                }
+            }
+          }
+        }
+     }
+  }
+}
+EOF
+)
 
 echo '' ; echo '==> Installing syslog output plugin...'
 
 /var/vcap/packages/logstash/logstash/bin/plugin install logstash-output-syslog
 
-
 echo '' ; echo '==> Shipping test data...'
 
 /var/vcap/packages/logstash/logstash/bin/logstash --config config/repo-commits.conf
 
+echo '' ; echo '==> Waiting for shipped logs to be ingested and indexed...'
 
-echo '' ; echo '==> Waiting for sanity...'
-
-sleep 5
-
+n=0
+until [ $n -ge 5 ]
+do
+  curl -s -d "$QUERY" -X POST '<%= p('test_e2e_errand.elasticsearch_hostport') %>/_count' > curl.out
```

Comment:
Originally, I was always being as specific as possible - rather than `/_count`, consider doing `/_all/sample-logs--repo-commits/_count` so you also end up with an implicit test on the type you're expecting.